homeassistant/components/netgear/__init__.py

```diff
@@ -52,6 +55,34 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
         configuration_url=f"http://{entry.data[CONF_HOST]}/",
     )
 
+    async def async_update_data() -> None:
+        """Fetch data from the router."""
+        await router.async_update_device_trackers()
+
+    # Create update coordinator
+    coordinator = DataUpdateCoordinator(
+        hass,
+        _LOGGER,
+        name=router.device_name,
+        update_method=async_update_data,
+        update_interval=SCAN_INTERVAL,
+    )
+
+    # ensure the coordinator keeps updating to discover new devices
+    @callback
+    def dummy_coordinator_callback() -> None:
+        """Callback ensuring discovery of new devices."""
+
+    remove_dummy_update = coordinator.async_add_listener(dummy_coordinator_callback)
```

Comment:
Instead of adding a dummy listener I suggest the default listener should be the callback for new devices that creates new entities.

Reply:
done, tested and it works.